src/modules/solution-explorer/solution-explorer-solution/delete-diagram-modal/delete-diagram-modal.ts

```diff
@@ -0,0 +1,65 @@
+import {inject} from 'aurelia-framework';
+
+import {IDiagram} from '@process-engine/solutionexplorer.contracts';
+import {ISolutionExplorerService} from '@process-engine/solutionexplorer.service.contracts';
+
+import {NotificationType} from '../../../../contracts/index';
+import {NotificationService} from '../../../notification/notification.service';
+
+@inject('NotificationService')
+export class DeleteDiagramModal {
+  public showModal: boolean = false;
+
+  private _diagram: IDiagram;
+  private _solutionService: ISolutionExplorerService;
+  private _notificationService: NotificationService;
+
+  constructor(notificationService: NotificationService) {
+    this._notificationService = notificationService;
+  }
+
+  public async show(diagram: IDiagram, solutionService: ISolutionExplorerService): Promise<boolean> {
+    this._diagram = diagram;
+    this._solutionService = solutionService;
+
+    this.showModal = true;
+
+    const deletionPromise: Promise<boolean> = new Promise((resolve: Function, reject: Function): boolean | void => {
```

Comment:
```suggestion
    const deletionPromise: Promise<boolean> = new Promise((resolve: Function, reject: Function): void => {
```

The callback function itself doesn't return anything, so `void` should suffice here.